src/javascript/SelectorTypes/Picker/PickerDialog/RightPanel/RightPanel.jsx

```diff
@@ -29,12 +41,47 @@ const RightPanel = ({pickerConfig, onClose, onItemSelection}) => {
 
     const viewSelector = registry.get('accordionItem', mode)?.viewSelector;
 
+    const handleChangeTerms = e => {
+        console.log('Updating search terms', e.target.value);
+        dispatch(cePickerSetSearchTerm(e.target.value));
+    };
+
+    const handleClearTerms = e => {
+        console.log('Clearing search terms', e.target.value);
+        dispatch(cePickerSetSearchTerm(''));
+        dispatch(cePickerSetSearchContext(currentPath));
+    };
+
+    const handleChangeContext = (e, item) => {
+        console.log('Updating search context', item.value);
+        dispatch(cePickerSetSearchContext(item.value));
+    };
+
+    const getSearchContextData = () => {
+        return pickerConfig.pickerDialog.searchContextOptions(currentPath, currentSite, t).filter((value, index, array) => array.findIndex(value1 => value1.value === value.value) === index);
+    };
+
+    const getCurrentSearchContext = () => {
+        return getSearchContextData().find(value => value.value === (searchContext === '' ? currentPath : searchContext));
+    };
+
     return (
         <div className="flexFluid flexCol_nowrap">
             <header className={clsx('flexCol_nowrap', css.header)}>
                 <Typography variant="heading">Select content</Typography>
                 <div className={clsx('flexRow_nowrap', 'alignCenter', css.headerActions)}>
-                    <SearchContextInput size="big" className={css.searchInput}/>
+                    <SearchContextInput
```

Comment:
Could be refactored to its own component as it looks independent enough just to avoid clutter (optional)

Reply:
Did not refactor in a sub component